Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -1426,41 +1426,75 @@ static void webProcessPoolHighDynamicRangeDidChangeCallback(CFNotificationCenter
 }
 #endif
 
-static void addUserInstalledFontURLs(NSString *path, Vector<URL>& fontURLs)
+static Vector<SandboxExtension::Handle> sandboxExtensionsForUserInstalledFonts(const Vector<URL>& fontPathURLs, std::optional<audit_token_t> auditToken)
+{
+    Vector<SandboxExtension::Handle> handles;
+    for (auto& fontPathURL : fontPathURLs) {
+        std::optional<SandboxExtension::Handle> sandboxExtensionHandle;
+        if (auditToken)
+            sandboxExtensionHandle = SandboxExtension::createHandleForReadByAuditToken(fontPathURL.fileSystemPath(), *auditToken);
+        else
+            sandboxExtensionHandle = SandboxExtension::createHandle(fontPathURL.fileSystemPath(), SandboxExtension::Type::ReadOnly);
+        if (sandboxExtensionHandle)
+            handles.append(WTFMove(*sandboxExtensionHandle));
+    }
+    return handles;
+}
+
+static void addUserInstalledFontURLs(NSString *path, HashMap<String, URL>& fontURLs, Vector<URL>& sandboxExtensionURLs)
 {
+    RetainPtr pathURL = [NSURL fileURLWithPath:path];
+    sandboxExtensionURLs.append(URL(pathURL.get()));
+
     RetainPtr enumerator = [NSFileManager.defaultManager enumeratorAtPath:path];
 
     for (NSString *font in enumerator.get()) {
         RetainPtr nsFontURL = [NSURL fileURLWithPath:[path stringByAppendingPathComponent:font]];
         RetainPtr utType = [UTType typeWithFilenameExtension:nsFontURL.get().pathExtension];
         if ([utType isSubtypeOfType:UTTypeFont]) {
             URL fontURL(nsFontURL.get());
-            fontURLs.append(fontURL);
-            RELEASE_LOG(Process, "Registering font url %s", fontURL.string().utf8().data());
+            auto fontDescriptors = adoptCF(CTFontManagerCreateFontDescriptorsFromURL((__bridge CFURLRef)nsFontURL.get()));
+            for (CFIndex i = 0; i < CFArrayGetCount(fontDescriptors.get()); ++i) {
+                RetainPtr fontDescriptor = static_cast<CTFontDescriptorRef>(CFArrayGetValueAtIndex(fontDescriptors.get(), i));
+                if (!fontDescriptor)
+                    continue;
```

Comment:
Should we update sandboxExtensionURLs only if `fontURL` is added to map `fontURLs`? So if the fontURL does not make into the font map the we don't need to create sandbox extension for it (and in this case `sandboxExtensionURLs` should probably be a map)